Show error message only if there was an error

RonEld opened the pull request:
This resolves #852. Print the error message, in `key_app`
only if there was an error.

sbutcher-arm commented on programs/pkey/key_app.c:
```diff
@@ -265,8 +265,11 @@ int main( int argc, char *argv[] )
 #if defined(MBEDTLS_ERROR_C)
-    mbedtls_strerror( ret, buf, sizeof(buf) );
-    mbedtls_printf( "  !  Last error was: %s\n", buf );
+    if ( ret != 0 )
```
Just before this check to `ret` there are multiple calls to `mbedtls_mpi_write_file`, none of which check their return code. Can you please expand the error checking to the other library calls?

RonEld replied:
Hi @sbutcher-arm I fixed according to your comment

andresag01 commented on programs/pkey/key_app.c:
```diff
@@ -239,34 +239,37 @@ int main( int argc, char *argv[] )
-    mbedtls_printf( "  !  Last error was: %s\n", buf );
+    if ( ret != 0 )
+    {
+        mbedtls_strerror( ret, buf, sizeof(buf) );
```
Style: Please add space before and after `)` and `(`.

sbutcher-arm commented:
Somehow this has conflicts. @RonEld - Could you please rebase? Thanks.

RonEld commented:
The conflicts arose from commit 54ebf9971d1645b2f08623df4b9e40204c4ffd61

@sbutcher-arm I have rebased and resolved conflict
@andresag01 I have fixed the preexisting style issue

sbutcher-arm commented:
I think we should backport this, as the behaviour in the maintenance branches is simply wrong. @roneld - can you either provide backports or give some justification of why we shouldn't?

cc: @Patater

RonEld commented:
@sbutcher-arm I don't mind backporting this. We usually don't backport sample application, especially that this bug is just an empty error message, when there is no error.

RonEld commented:
Backports available at #1799 and #1800 

I have rebased and fixed compilation errors in CI due to new code introduced.
@sbutcher-arm @andresag01 please review

sbutcher-arm commented:
Note, this looks like it'll conflict with #532, which as a larger PR, I'd like to merge first.

RonEld commented:
@andresag01 I am not sure I am following you.
I haven't removed any code. Only added the `MBEDTLS_MPI_CHK`.
`mbedtls_export()` and `mbedtls_export_crt()` is still called, but sooner.
The changes you are referring to were introduced in https://github.com/ARMmbed/mbedtls/commit/54ebf9971d1645b2f08623df4b9e40204c4ffd61 , ao I think @hanno-arm can give a better answer

andresag01 commented:
@RonEld: According to the diff, commit 6a9257bc5719dcdcc44dd2c0f52208012b1bffe9 from this PR makes the following kind of changes:
Original:
```
mbedtls_mpi_write_file( "N:  ",  &N,  16, NULL );
```
New:
```
MBEDTLS_MPI_CHK( mbedtls_mpi_write_file( "N:  ", &rsa->N, 16, NULL ) );
```
Note that in the `New` version, we no longer have `&N` but `&rsa->N`, which means that the example is manually inspecting the context of the RSA key instead of using the output of the export functions, which I believe its effectively redundant with the change I pointer out above.

RonEld commented:
Oh, I see what you mean, Thanks!
It was a merge conflict, I'll fix it ASAP

RonEld commented:
@andresag01 I have fixed this, please review

RonEld commented:
@andresag01 Thanks for your approval. You haven't spotted this so late, as it was introduced only today, with the rebase, and it was a difficult one to spot!

sbutcher-arm commented:
The CI is failing because the FreeBSD build is complaining it can't find `scripts/config.pl`, which is an odd failure. If it went missing I'd expect lots of other, earlier failures, therefore re-running the CI.

sbutcher-arm commented:
retest

sbutcher-arm commented:
Needs a Changelog entry, but that can be done at Gatekeeping.
